src/components/templates/PartyMap/components/Map/PartyMapRoomOverlay.tsx

```diff
@@ -14,104 +17,205 @@ import RoomAttendance from "../RoomAttendance";
+    //   prevRoomClicked === room.title ? undefined : room.title
+    // );
+    // }
   }, [room, selectRoom]);
 
-  const [isRoomHovered, setRoomHovered] = useState<
-    PartyMapRoomData | undefined
-  >(undefined);
+  // TODO: this was used in place of onClick1 in PartyMap Map before I merged them, is it correct? Or is the onClick1 definition above?
+  // const toggleRoom = useCallback(() => {
+  //   selectRoom(room);
+  // }, [room, selectRoom]);
+
+  const [roomHovered, setRoomHovered] = useState<PartyMapRoomData | undefined>(
+    undefined
+  );
+
+  const isRoomHovered = isTruthy(roomHovered);
 
   const dispatch = useDispatch();
 
-  const roomHovered = useCallback(() => {
-    dispatch(retainAttendance(true));
+  const handleRoomHovered = useCallback(() => {
     setRoomHovered(room);
+    dispatch(retainAttendance(true));
   }, [dispatch, room]);
 
-  const roomUnhovered = useCallback(() => {
-    dispatch(retainAttendance(false));
+  const handleRoomUnhovered = useCallback(() => {
     setRoomHovered(undefined);
+    dispatch(retainAttendance(false));
   }, [dispatch]);
 
-  const left = room.x_percent;
-  const top = room.y_percent;
-  const width = room.width_percent;
-  const height = room.height_percent;
+  // TODO: this is being spread here because the below code modifies it with .push()/etc
+  // const rooms = [...venue.rooms];
+
+  // TODO: what is the intent/purpose of this?
+  // TODO: how do we implement this nicely, and where?
+  // TODO: reinstate this functionality
+  // if (roomHovered) {
+  //   const idx = rooms.findIndex((room) => room.title === roomHovered.title);
+  //   if (idx !== -1) {
+  //     const chosenRoom = rooms.splice(idx, 1);
+  //     rooms.push(chosenRoom[0]);
+  //   }
+  // }
+
+  // TODO: what is a better semantic name for this?
+  // const onJoinRoom: React.MouseEventHandler = useCallback(
+  //   (e) => {
+  //     e.stopPropagation();
+  //
+  //     if (isExternalUrl(room.url)) {
+  //       openRoomUrl(room.url);
+  //     } else {
+  //       window.location.href = getRoomUrl(room.url);
+  //     }
+  //
+  //     enterCampRoom(room);
+  //   },
+  //   [enterCampRoom, room]
+  // );
+
+  // TODO: should this also openRoomUrl as above? If so, let's use the same function for it.. if not, why?
+  // TODO: what is a better semantic name for this?
+  // const onClick3 = useCallback(() => enterCampRoom(room), [
+  //   enterCampRoom,
+  //   room,
+  // ]);
+
+  // TODO: what is a better semantic name for this?
+  const shouldShowHovered =
+    venue.roomVisibility === RoomVisibility.hover &&
+    isRoomHovered &&
+    roomHovered?.title === room.title;
+
+  // TODO: what is a better semantic name for this?
+  const shouldShow1 =
+    !venue.roomVisibility || venue.roomVisibility === RoomVisibility.nameCount;
+
+  // TODO: what is a better semantic name for this?
+  const shouldShow2 =
+    shouldShow1 ||
+    (venue.roomVisibility === RoomVisibility.count && hasAttendance);
```

Comment:
- [ ] [TODO] what is a better semantic name for this?

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/840/commits/4b9e920e8fc98fbba69c616e495c9d2e9ec33a2a